File: 프로그래머스/lv2/12914. 멀리 뛰기/멀리 뛰기.kt
class Solution {
    fun solution(n: Int): Long {
        var answer: Long = 0
        var d = LongArray(n+1)
        if(n == 1)
        return 1
        if(n == 2)
        return 2
        
        d[0] = 0
        d[1] = 1L
        d[2] = 2L
        
        for(i in 3..n){
            d[i] = (d[i-1] + d[i-2]) % 1234567
        }
        
        
    
        return d[n]
    }
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.js
function solution(n) {
    let answer = 0;
    let MOD = 1234567;
    
    let dp = new Array(n+1 ).fill(0);
    
    dp[0] = 1;
    dp[1] = 1;
    
    for(let i = 2; i <= n; i++){
        dp[i] = (dp[i-1] + dp[i-2]) % MOD;
     }
    
    
    return dp[n];
}